File: python/obtain-auth-token.py
#!/usr/bin/env python
"""
Post seafile email/password to obtain auth token.
"""

import urllib
import urllib2
import simplejson as json


url = 'https://seacloud.cc/api2/auth-token/'
values = {'username': 'demo@seafile.com',
          'password': 'demo'}
data = urllib.urlencode(values)
req = urllib2.Request(url, data)
response = urllib2.urlopen(req)
the_page = response.read()
token = json.loads(the_page)['token']

print token

File: javascript/obtain-auth-token.js
var invocation = new XMLHttpRequest();
var url = 'https://seacloud.cc/api2/auth-token/';
function getSeafileApiRequest() {
    if(invocation) {    
        invocation.open('POST', url, true);
        invocation.setRequestHeader("Content-type", "application/x-www-form-urlencoded");
        invocation.onreadystatechange = function() {
            if(invocation.readyState == 4 && invocation.status == 200) {
                console.log(invocation.responseText);
            }
        };

        invocation.send("username=demo@seafile.com&password=demo");
    }
}

getSeafileApiToken();
